src/autoloads/resources.gd

```diff
@@ -0,0 +1,71 @@
+extends Node
+
+## HOW TO USE THE RESOURCE MANAGER
+## Example: get a list of resources in a set of directories
+## Dictionary of directories, with the namespace, followed by the directory path:
+#	var dirs: Dictionary = {
+#	"Resource 1": "res://examples/resource1",
+#	"Resource 2": "res://examples/resource2"
+#	}
+#
+## Array of file types to use (only files with these extensions will be listed):
+#	var file_types: Array = [".png", ".svg"]
+#
+## Pass the two variables into the list function:
+#	Resources.list(dirs, file_types)
+#	> {Resource 1:{Test 1:test_1.png, Test 2:test_2.svg}, Resource 2:{Test 3:test_3.svg}}
+#
+## Select only resources in "Resource 1":
+#	Resources.list(dirs, file_types)["Resource 1"]
+#	> {Test 1:test_1.png, Test 2:test_2.svg}
+
+# --Public Functions--
+
+# Returns a dictionary of all resources from a given directory dictionary
+func list(directories: Dictionary, types: Array) -> Dictionary:
+	var resources: Dictionary = {} # Initialise resources dictionary
+	for folder in directories.keys(): # Iterate over each folder specified, by their namespace (key)
+		var files = _listFilesInDirectory(directories[folder], types) # List files in each folder
+		# Add each file to the output dictionary
+		for file in files :
+			resources = _filesToDictionary(resources, folder, file, directories[folder], types)
```

Comment:
This is kind of weird, `_filesToDictionary()` edits the `resources` dictionary already so why set `resources` again? Why not just have `_filesToDictionary()` create an empty dictionary itself instead of passing it one?

Reply:
I think I've got it like that because the _filesToDictionary is passed through a loop. But I'm gonna change it so it is passed the files array, rather than an individual file, which should fix this hopefully.

Reply:
hopefully this time it is better